### ISSUE TITLE
error: no matching function for call to 'simit::util::ScopedMap<simit::ir::Var, llvm::Value*>::insert(const simit::ir::Var&, llvm::ilist_iterator<llvm::Argument>&)'

### ISSUE THREAD

**Issue by ignatenkobrain** [simit-lang/simit#28]
```cxx
/home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/backend/llvm/llvm_backend.cpp: In member function 'llvm::Function* simit::backend::LLVMBackend::emitEmptyFunction(const string&, const std::vector<simit::ir::Var>&, const std::vector<simit::ir::Var>&, bool, bool, bool)':
/home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/backend/llvm/llvm_backend.cpp:1578:43: error: no matching function for call to 'simit::util::ScopedMap<simit::ir::Var, llvm::Value*>::insert(const simit::ir::Var&, llvm::ilist_iterator<llvm::Argument>&)'
     symtable.insert(*simitArgIt, llvmArgIt);
                                           ^
In file included from /home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/backend/llvm/llvm_backend.h:15:0,
                 from /home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/backend/llvm/llvm_backend.cpp:1:
/home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/util/scopedmap.h:41:8: note: candidate: void simit::util::ScopedMap<Key, Value>::insert(const Key&, const Value&) [with Key = simit::ir::Var; Value = llvm::Value*]
   void insert(const Key &symbol, const Value &value) {
        ^~~~~~
/home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/util/scopedmap.h:41:8: note:   no known conversion for argument 2 from 'llvm::ilist_iterator<llvm::Argument>' to 'llvm::Value* const&'
/home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/util/scopedmap.h:46:8: note: candidate: void simit::util::ScopedMap<Key, Value>::insert(const std::pair<_T1, _T2>&) [with Key = simit::ir::Var; Value = llvm::Value*]
   void insert(const std::pair<Key, Value> &symVal) {
        ^~~~~~
/home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/util/scopedmap.h:46:8: note:   candidate expects 1 argument, 2 provided
/home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/backend/llvm/llvm_backend.cpp:1583:43: error: no matching function for call to 'simit::util::ScopedMap<simit::ir::Var, llvm::Value*>::insert(const simit::ir::Var&, llvm::ilist_iterator<llvm::Argument>&)'
     symtable.insert(*simitResIt, llvmArgIt);
                                           ^
In file included from /home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/backend/llvm/llvm_backend.h:15:0,
                 from /home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/backend/llvm/llvm_backend.cpp:1:
/home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/util/scopedmap.h:41:8: note: candidate: void simit::util::ScopedMap<Key, Value>::insert(const Key&, const Value&) [with Key = simit::ir::Var; Value = llvm::Value*]
   void insert(const Key &symbol, const Value &value) {
        ^~~~~~
/home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/util/scopedmap.h:41:8: note:   no known conversion for argument 2 from 'llvm::ilist_iterator<llvm::Argument>' to 'llvm::Value* const&'
/home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/util/scopedmap.h:46:8: note: candidate: void simit::util::ScopedMap<Key, Value>::insert(const std::pair<_T1, _T2>&) [with Key = simit::ir::Var; Value = llvm::Value*]
   void insert(const std::pair<Key, Value> &symVal) {
        ^~~~~~
/home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/util/scopedmap.h:46:8: note:   candidate expects 1 argument, 2 provided
src/CMakeFiles/simit.dir/build.make:1289: recipe for target 'src/CMakeFiles/simit.dir/backend/llvm/llvm_backend.cpp.o' failed
make[2]: *** [src/CMakeFiles/simit.dir/backend/llvm/llvm_backend.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/backend/llvm/llvm_function.cpp: In member function 'virtual simit::backend::Function::FuncType simit::backend::LLVMFunction::init()':
/home/brain/rpmbuild/BUILD/simit-6083206fcd97909daf04bf2bc5968363feb31136/src/backend/llvm/llvm_function.cpp:344:45: error: cannot convert 'llvm::ilist_iterator<llvm::Argument>' to 'llvm::Argument*' in initialization
       llvm::Argument* llvmFormal = llvmArgIt++;
                                             ^~
```

llvm-3.8.0-1.fc25.x86_64

**Comment by zyv** [comment]
Duplicate of #21.

**Comment by fredrikbk** [comment]
Closed as duplicate of #21
